pulp_container/app/registry_api.py

```diff
@@ -473,6 +477,135 @@ def get_queryset(self, *args, **kwargs):
+        if architectures:
+            mlms.filter(architecture__in=architectures)
+        for mlm in mlms:
+            self.recurse_through_manifest_lists(
+                tag, mlm.manifest_list, oss, architectures, manifests
+            )
+        if manifest.media_type in (models.MEDIA_TYPE.MANIFEST_V2, models.MEDIA_TYPE.MANIFEST_OCI):
+            manifests.setdefault(manifest, set()).add(tag)
+
+    def get(self, request):
+        req_repositories = None
+        req_tags = None
+        req_oss = None
+        req_architectures = None
+        req_label_exists = set()
+        req_label_values = {}
+        for key, values in request.query_params.lists():
+            if key == "repository":
+                req_repositories = values
+            elif key == "tag":
+                req_tags = values
+            elif key == "os":
+                req_oss = values
+            elif key == "architecture":
+                req_architectures = values
+            elif key.startswith("label:"):
+                if key.endswith(":exists"):
+                    if any(v != "1" for v in values):
+                        raise ParseError()
+                    label = key[len("label:") : len(key) - len(":exists")]
+                    req_label_exists.add(label)
+                else:
+                    label = key[len("label:") :]
+                    req_label_values[label] = values
+            else:
+                # In particularly, this covers any annotation:... parameters, which this
+                # implementation does not support:
+                raise ParseError()
+
+        if "org.flatpak.ref" not in req_label_exists:
+            raise ParseError()
+
+        distributions = models.ContainerDistribution.objects.filter(private=False).only("base_path")
+
+        if req_repositories:
+            distributions = distributions.filter(base_path__in=req_repositories)
+
+        results = []
+        for distribution in distributions:
+            images = []
+            if distribution.repository:
+                repository_version = distribution.repository.latest_version()
+            else:
+                repository_version = distribution.repository_version
+            tags = models.Tag.objects.filter(pk__in=repository_version.content)
+            if req_tags:
+                tags = tags.filter(name__in=req_tags)
+            manifests = {}  # mapping manifests to sets of encountered tag names
+            for tag in tags:
+                self.recurse_through_manifest_lists(
+                    tag.name, tag.tagged_manifest, req_oss, req_architectures, manifests
+                )
+            for manifest, tagged in manifests.items():
+                with storage.open(manifest.config_blob._artifacts.get().file.name) as file:
+                    raw_data = file.read()
+                config_data = json.loads(raw_data)
+                labels = config_data.get("config", {}).get("Labels")
```

Comment:
```suggestion
                labels = config_data.get("config", {}).get("Labels", [])
```

`labels` can be `None` here. On line 585 this might not be appropriate, right? If the protocol does not define its behaviour, we can omit the entry.

Reply:
should be fine, see the following `if not labels:  continue`